tests/integration/cli/client_index_list_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package clitest
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type indexFieldResp struct {
+	Name string `json:"Name"`
+}
+type indexResp struct {
+	Name   string           `json:"Name"`
+	Fields []indexFieldResp `json:"Fields"`
+}
+
+func TestIndexList_IfCollectionIsNotSpecified_ShouldReturnAllIndexes(t *testing.T) {
+	conf := NewDefraNodeDefaultConfig(t)
+	stopDefra := runDefraNode(t, conf)
+
+	createCollection(t, conf, `type User { Name: String }`)
+	createCollection(t, conf, `type Product { Name: String Price: Int }`)
```

Comment:
todo: We agreed to follow the convention of `Type` and `field` for our schemas. This should be changes to `type Product { name: String, price: Int }`.